src/reporters/json.ts

```diff
@@ -27,18 +27,119 @@ import BaseReporter from './base';
 import { formatError, getTimestamp } from '../helpers';
 import { Journey, Step } from '../dsl';
 import snakeCaseKeys from 'snakecase-keys';
+import { NetworkInfo } from '../common_types';
 
 /* eslint-disable @typescript-eslint/no-var-requires */
-const programVersion = require('../../package.json').version;
+const { version, name } = require('../../package.json');
+
+type OutputType =
+  | 'journey/register'
+  | 'journey/start'
+  | 'step/screenshot'
+  | 'step/end'
+  | 'journey/network_info'
+  | 'journey/filmstrips'
+  | 'journey/browserconsole'
+  | 'journey/end';
+
+type OutputFields = {
+  type: OutputType;
+  journey: Journey;
+  timestamp?: number;
+  url?: string;
+  step?: Partial<Step>;
+  error?: Error;
+  root_fields?: Record<string, unknown>;
+  payload?: Record<string, unknown>;
+  blob?: string;
+};
+
+function getMetadata() {
+  return {
+    process: {
+      pid: process.pid,
+      ppid: process.ppid,
+      title: process.title,
+      args: process.argv,
+    },
+    os: {
+      platform: process.platform,
+    },
+    package: {
+      name,
+      version,
+    },
+  };
+}
+
+function formatHTTPVersion(protocol: string) {
+  if (protocol === 'h2') {
+    return 2;
+  } else if (protocol === 'http/1.1') {
+    return 1.1;
+  } else if (protocol === 'http/1.0') {
+    return 1.0;
+  } else if (protocol && protocol.startsWith('h3')) {
+    return 3;
+  }
+  return;
+}
+
+export function formatNetworkFields(network: NetworkInfo) {
+  const { request, response, url } = network;
+  const postdata = request.postData || '';
+  const tls = response?.securityDetails;
+
+  return {
+    // URL and USER AGENT would be parsed and mapped by heartbeat
+    url,
+    user_agent: request.headers?.['User-Agent'],
+    http: {
+      version: formatHTTPVersion(response?.protocol),
+      request: {
+        body: {
+          bytes: postdata.length,
+          content: postdata,
+        },
+        method: request.method,
+        referrer: request.headers?.Referer,
+      },
+      response: response
+        ? {
+            body: {
+              bytes: response.encodedDataLength,
+            },
+            mime_type: response.mimeType,
+            status_code: response.status,
+          }
+        : undefined,
+    },
+    tls: tls
```

Comment:
We should have this match what heartbeat does more closely, specifically nesting under `tls.server` instead of `tls.client` and enumerating more fields if possible. The `subject` field is particularly imporant.

Here's an example of the fields heartbeat provides.

```json
{
  "cipher": "ECDHE-RSA-AES-128-GCM-SHA256",
  "server": {
    "x509": {
      "subject": {
        "common_name": "r2.shared.global.fastly.net",
        "distinguished_name": "CN=r2.shared.global.fastly.net,O=Fastly\\, Inc.,L=San Francisco,ST=California,C=US"
      },
      "serial_number": "25044290308248398422640150363",
      "public_key_size": 2048,
      "public_key_exponent": 65537,
      "not_before": "2021-02-05T17:27:13.000Z",
      "issuer": {
        "common_name": "GlobalSign CloudSSL CA - SHA256 - G3",
        "distinguished_name": "CN=GlobalSign CloudSSL CA - SHA256 - G3,O=GlobalSign nv-sa,C=BE"
      },
      "not_after": "2021-06-20T21:11:25.000Z",
      "signature_algorithm": "SHA256-RSA",
      "public_key_algorithm": "RSA"
    },
    "hash": {
      "sha256": "bf2b471bc6d729f0b0359bbaaf7cead9c1dffd7b8e7a15ea136e7468ef8b7d87",
      "sha1": "e005c41f94da793d7f812c092e1a432b6c58498e"
    }
  },
  "established": true,
  "rtt": {
    "handshake": {
      "us": 31066
    }
  },
  "version_protocol": "tls",
  "version": "1.2"
}
```

Reply:
Great point, will dig more and add these.